massa-consensus/src/settings.rs

```diff
@@ -157,7 +159,14 @@ pub struct ConsensusSettings {
 }
 
 impl ConsensusSettings {
+    /// Utility method to derivate a full configuration (static + user defined) from an user defined one
+    /// This is really handy in tests where we want to mutate static defined values
     pub fn config(&self) -> ConsensusConfig {
+        // TODO: these assertion should be checked at compile time
+        // https://github.com/rust-lang/rfcs/issues/2790
+        assert!(THREAD_COUNT >= 1);
```

Comment:
Clippy is wrong there ?

Reply:
No is right and that's really good news :)